src/components/SelectableVisualization.vue

```diff
@@ -259,9 +270,39 @@ export default {
       }
       return date;
     },
+    timeline_daterange: function () {
+      let date = this.activityStore.query_options.date;
+      if (!date) {
+        date = this.activityStore.query_options.timeperiod.start;
+      }
+
+      // Get events data from CurrentData ± 1 Day so that there is some continuity
+      return [moment(date).subtract(1, 'day'), moment(date).add(1, 'day')];
+    },
     isSingleDay: function () {
       return _.isEqual(this.activityStore.query_options.timeperiod.length, [1, 'day']);
     },
   },
+  watch: {
+    timeline_daterange: async function () {
+      await this.getTimelineBuckets();
+    },
+  },
+  mounted: async function () {
+    await this.getTimelineBuckets();
+  },
```

Comment:
This should only happen if the vis_timeline component is actually used, and then only once for a given set of view parameters and not once-per-`SelectableVisualization` component (currently prob spams requests?).

Might want to write a small wrapper around vis-timeline to do the needed fetching there? Or some other solution, haven't thought about it too deeply.

Reply:
Nice catch ! I can "type" check the current visualization then conditionally getBuckets ?! Next update occur in computed daterange change (view params changes as expected)

Since vis-timeline is already core component, didn't write a wrapper around it :3

Reply:
Ah yes ofc that could work. Still a kind of messy place to put the code in, but I could fix that later.